docs/design/include/mcmini/shared_sem.h

```diff
@@ -0,0 +1,29 @@
+#ifndef INCLUDE SHARE_SEM_H
+#define INCLUDE_SHARE_SEM_H
+
+#include <semaphore.h>
+#define MC_STRUCT_DECL(type)       \
+  typedef struct type type;        \
+  typedef struct type *type##_t;   \
+  typedef struct type *type##_ref; \
+  typedef const struct type *type##_refc;
+
+MC_STRUCT_DECL(shared_sem)
+struct shared_sem {
+  sem_t runner_sem; //previously named dpor_scheduler_sem
+  sem_t pthread_sem;
```

Comment:
See my comment about the functions wrapping these. Using `runner` and `thread` both is confusing because `runner` refers to the thread.